modules/statistics/php/NDB_Menu_statistics_site.class.inc

```diff
@@ -23,7 +23,16 @@
  */
 class NDB_Menu_Statistics_Site extends NDB_Menu
 {
-
+    /**
+     * Checking user's permission
+     *
+     * @return bool
+     */
+    function _hasAccess()
+    {
+        $user =& User::singleton();
+        return $user->hasPermission('data_entry');
+    }
```

Comment:
The _checkCriteria function looks like it modifies the results that are returned to ensure they don't see other site's candidates in the results after they have permission, while the hasAccess would modify whether the user gets a permission denied error when trying to access the page.